config/ssl/dynamo/dynamo.adb

```diff
@@ -0,0 +1,314 @@
+--  later version. This library is distributed in the hope that it will be  --
+--  useful, but WITHOUT ANY WARRANTY;  without even the implied warranty of --
+--  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.                    --
+--                                                                          --
+--  As a special exception under Section 7 of GPL version 3, you are        --
+--  granted additional permissions described in the GCC Runtime Library     --
+--  Exception, version 3.1, as published by the Free Software Foundation.   --
+--                                                                          --
+--  You should have received a copy of the GNU General Public License and   --
+--  a copy of the GCC Runtime Library Exception along with this program;    --
+--  see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see   --
+--  <http://www.gnu.org/licenses/>.                                         --
+--                                                                          --
+--  As a special exception, if other files instantiate generics from this   --
+--  unit, or you link this unit with other files to produce an executable,  --
+--  this  unit  does not  by itself cause  the resulting executable to be   --
+--  covered by the GNU General Public License. This exception does not      --
+--  however invalidate any other reasons why the executable file  might be  --
+--  covered by the  GNU Public License.                                     --
+------------------------------------------------------------------------------
+
+with Ada.Command_Line;      use Ada.Command_Line;
+with Ada.Strings.Wide_Wide_Unbounded;
+with Ada.Text_IO;
+with Ada.Wide_Wide_Characters.Handling;
+
+with Langkit_Support.Text;
+
+with Libadalang.Analysis;   use Libadalang.Analysis;
+with Libadalang.Common;     use Libadalang.Common;
+
+function Dynamo return Integer is
+   package TIO renames Ada.Text_IO;
+   package WCH renames Ada.Wide_Wide_Characters.Handling;
+
+   Source_Path : constant String := Argument (1);
+   Main_Source : constant String := Argument (2);
+   Bind_Source : constant String := Argument (3);
+
+   Ctx  : constant Analysis_Context := Create_Context;
+   Unit : constant Analysis_Unit := Ctx.Get_From_File (Source_Path);
+
+   Charset : constant String := Get_Charset (Unit);
+
+   function WW_Argument (P : Positive) return Wide_Wide_String is
+     (Langkit_Support.Text.Decode (Argument (P), Charset));
+
+   Bind_Pkg   : constant Wide_Wide_String := WW_Argument (4);
+   Loader_Pkg : constant Wide_Wide_String := WW_Argument (5);
+
+   First : Token_Reference := Unit.First_Token;
+
+   WLF : constant Wide_Wide_Character :=
+           Wide_Wide_Character'Val (Character'Pos (ASCII.LF));
+
+   Counter : Positive := 1;
+
+   CU : constant Ada_Node := Unit.Root;
+   S  : TIO.File_Type;
+   B  : TIO.File_Type;
+
+   procedure Put (File : TIO.File_Type; Node : Ada_Node'Class);
```

Comment:
We need to comment all routines.
Generally the code is missing comments, can you do a pass and add comments to ensure easier maintenance later.

Reply:
All done. I hope dynamo.adb has enough comments now.